ui/selectors/selectors.js

```diff
@@ -1089,6 +1089,50 @@ export const getSnap = createDeepEqualSelector(
   },
 );
 
+/**
+ * Get a selector that returns the snap manifest for a given `snapId`.
+ *
+ * @param {object} state - The Redux state object.
+ * @param {string} snapId - The snap ID to get the manifest for.
+ * @returns {object | undefined} The snap manifest.
+ */
+export const getSnapManifest = createDeepEqualSelector(
+  (state) => state.metamask.currentLocale,
+  (state, snapId) => getSnap(state, snapId),
+  (locale, snap) => {
+    if (!snap?.localizationFiles) {
+      return snap?.manifest;
+    }
+
+    return getLocalizedSnapManifest(
+      snap.manifest,
+      locale,
+      snap.localizationFiles,
+    );
+  },
+);
+
+/**
+ * Get a selector that returns the snap metadata (name and description) for a
+ * given `snapId`.
+ *
+ * @param {object} state - The Redux state object.
+ * @param {string} snapId - The snap ID to get the metadata for.
+ * @returns {object} An object containing the snap name and description.
+ */
+export const getSnapMetadata = createDeepEqualSelector(
+  (state, snapId) => getSnapManifest(state, snapId),
+  (_, snapId) => snapId,
+  (manifest, snapId) => {
+    return {
+      // The snap manifest may not be available if the Snap is not installed, so
+      // we use the snap ID as the name in that case.
+      name: manifest?.proposedName ?? stripSnapPrefix(snapId),
```

Comment:
IIRC `snapId` is occasionally null when we get a too quick re-render. We may want to guard against that. See previous `getSnapName` function

Reply:
I don't see a guard against it in the previous function?

Reply:
That's what this is: https://github.com/MetaMask/metamask-extension/blob/201a48fd76e89c68ea9a559d10dc58fa14d7da61/ui/helpers/utils/util.js#L577-L580